File: api/predict.js
export default function handler(req, res) {
  if (req.method === "POST") {
    const { savings, goal, scheme } = req.body;

    // Simple dummy logic
    let color = "red";
    if (savings > 500000) color = "green";
    else if (savings > 250000) color = "orange";

    res.status(200).json({
      color,
      suggestion: `For your goal: ${goal} and scheme: ${scheme}, this is a ${color.toUpperCase()} signal.`,
    });
  } else {
    res.status(405).json({ message: "Method Not Allowed" });
  }
}

File: api/predict.py
# api/predict.py

from http.server import BaseHTTPRequestHandler
import json

class handler(BaseHTTPRequestHandler):
    def do_POST(self):
        content_length = int(self.headers['Content-Length'])
        post_data = self.rfile.read(content_length)
        data = json.loads(post_data)

        savings = int(data.get("savings", 0))
        goal = data.get("goal", "").lower()
        scheme = data.get("scheme", "").lower()

        if savings >= 500000:
            color = "green"
            tip = "You're financially ready."
        elif savings >= 200000:
            color = "orange"
            tip = "You may need to save more."
        else:
            color = "red"
            tip = "Savings too low for this goal."

        response = {
            "color": color,
            "tip": tip
        }

        self.send_response(200)
        self.send_header('Content-type', 'application/json')
        self.end_headers()
        self.wfile.write(json.dumps(response).encode())
